libs/application/templates/official-journal-of-iceland/src/hooks/useLastSignature.ts

```diff
@@ -0,0 +1,48 @@
+import { useLazyQuery, useQuery } from '@apollo/client'
+import { INVOLVED_PARTY_SIGNATURES_QUERY } from '../graphql/queries'
+import {
+  OfficialJournalOfIcelandApplicationInvolvedPartySignaturesCommittee,
+  OfficialJournalOfIcelandApplicationInvolvedPartySignaturesRegular,
+} from '@island.is/api/schema'
+
+type Props = {
+  involvedPartyId?: string
+}
+
+type LastSignaturesResponse = {
+  officialJournalOfIcelandApplicationGetInvolvedPartySignatures:
+    | OfficialJournalOfIcelandApplicationInvolvedPartySignaturesRegular
+    | OfficialJournalOfIcelandApplicationInvolvedPartySignaturesCommittee
+}
+
+export const useLastSignature = ({ involvedPartyId }: Props) => {
+  const { data, loading, error } = useQuery<LastSignaturesResponse>(
+    INVOLVED_PARTY_SIGNATURES_QUERY,
```

Comment:
Getur bætt við skip eigindinu hér til að sleppa því að kalla ef `involvedPartyId` er ekki sett.

```suggestion
    INVOLVED_PARTY_SIGNATURES_QUERY,
    skip: ! involvedPartyId,
```